### PR TITLE
feat: Make CatalogChunk first/last write times required

### PR DESCRIPTION
**Pull request by carols10cents**
This is the actual goal of https://github.com/influxdata/influxdb_iox/issues/1927, aside from some refactoring to see if we can make CatalogChunk the *only* place first/last write times are managed.

### CODE REVIEW

File: data_types/src/chunk_metadata.rs [hunk 1]
```diff
@@ -141,12 +140,12 @@ pub struct ChunkSummary {
     /// The earliest time at which data contained within this chunk was written
     /// into IOx. Note due to the compaction, etc... this may not be the chunk
     /// that data was originally written into
-    pub time_of_first_write: Option<DateTime<Utc>>,
+    pub time_of_first_write: DateTime<Utc>,
```
Comment (alamb) [inline data_types/src/chunk_metadata.rs:143]: 🎉

File: data_types/src/chunk_metadata.rs [hunk 2]
```diff
@@ -174,66 +173,14 @@ pub struct DetailedChunkSummary {
...
-
-    /// Normalizes a set of ChunkSummaries for comparison by removing timestamps
-    pub fn normalize_summaries(summaries: Vec<Self>) -> Vec<Self> {
-        let mut summaries = summaries
-            .into_iter()
-            .map(|summary| summary.normalize())
-            .collect::<Vec<_>>();
-        summaries.sort_unstable();
-        summaries
+    pub fn equal_without_timestamps(&self, other: &Self) -> bool {
```
Comment (alamb) [inline data_types/src/chunk_metadata.rs:176]: 👍 that is a nice change

**Review by alamb — APPROVED** [review]
This is so beautiful @carols10cents. Very nice. 


I wonder if we should change the descriptio of this PR to say it closes https://github.com/influxdata/influxdb_iox/issues/1927 ?

Or do you plan to remove TableSummaryWithTimestamps first?

File: lifecycle/src/lib.rs [hunk 3]
```diff
@@ -9,18 +9,19 @@
 )]
 
 use chrono::{DateTime, Utc};
-
-use data_types::chunk_metadata::{ChunkAddr, ChunkLifecycleAction, ChunkStorage};
```
Comment (crepererum) [inline lifecycle/src/lib.rs:13]: out of personal interest: are these manual import cleanups or are you using a tool for that?

**Review by crepererum — APPROVED** [review]

**Comment by carols10cents** [comment]
> I wonder if we should change the descriptio of this PR to say it closes #1927 ?
> 
> Or do you plan to remove TableSummaryWithTimestamps first?

I have one more PR coming that should hopefully simplify everything further :)

File: lifecycle/src/lib.rs (hunk 3 above)
Comment (carols10cents) [reply lifecycle/src/lib.rs:13]: Removing the newlines and grouping by crate is manual, then rustfmt handles the alphabetization, but I think I just found a rustfmt option that does what I like :) Going to see how much noise it makes in a separate PR....

Comment (tustvold) [reply lifecycle/src/lib.rs:13]: Are they by any chance

```
imports_granularity = "Crate"
group_imports = "StdExternalCrate"
```

I added these aspirationally to rustfmt.toml a while back but they needed nightly at the time

Comment (carols10cents) [reply lifecycle/src/lib.rs:13]: UGH yeah just noticed that. They're still nightly :(

**Review by tustvold — APPROVED** [review]
LGTM - I was reviewing when merged so figured I'd finish it up :laughing:

**Pull request merged**